library/src/main/java/com/novoda/downloadmanager/lib/Criteria.java

```diff
@@ -0,0 +1,177 @@
+package com.novoda.downloadmanager.lib;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class Criteria {
+
+    public static final Criteria ALL = new Criteria(null, null);
+
+    private final String selection;
+    private final String sort;
+    private final String[] selectionArguments;
+
+    Criteria(String selection, String sort, String... selectionArguments) {
+        this.selection = selection;
+        this.sort = sort;
+        this.selectionArguments = selectionArguments;
+    }
+
+    public String getSelection() {
+        return selection;
+    }
+
+    public String[] getSelectionArguments() {
+        return Arrays.copyOf(selectionArguments, selectionArguments.length);
+    }
+
+    public String getSort() {
+        return sort;
+    }
+
+    public static class Builder {
```

Comment:
The idea was to use the `Criteria` as a helper to construct the basic forms of a query in terms of selection, arguments, sort order. 

The `BatchQuery` is our public API where we define different filtering options. I don't know if it should be in sync with `Criteria`. 

Maybe in order to keep some consistency, the `Query` could use a similar approach, that of using the `Criteria` to construct the actual selections, and use a `repository` to do the actual querying as, at the moment, the `query` does too much (constructs the query and also runs it).